File: BCC221-POO/TPC++/classes_cpp/ordem.cpp
using namespace std;

#include "../headers/ordem.hpp"


Ordem::Ordem(string PedCliente, int id_cliente) {
    this->PedCliente = PedCliente;
    this->RepMec = "";
    this->orca = 0.0;
    this->status = false;
    this->id_cliente = id_cliente;

}

string Ordem::getPed(){
    return this->PedCliente;
}

string Ordem::getRepMec(){
    return this->RepMec;
}

string Ordem::getStatus(){

    if (this->status == true)
    {
        return "Aprovado";
    }

    else
    {
        return "Pendente";
    }
    
    
}

float Ordem::getOrca(){
    return this->orca;
}

void Ordem::setStatus(bool value){
    this->status = value;
}

void Ordem::setOrca(float value){
    this->orca = value;
}

void Ordem::setRepMec(string resposta){
    this->RepMec = resposta;
}

int Ordem::getIdCliente(){
    return this->id_cliente;
}

File: BCC221-POO/TPJAVA/TPJava/src/tpjava/Ordem.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package tpjava;

/**
 *
 * @author nicolas
 */
public class Ordem {
    private String PedOrdem; //V
    private String RepMec; //M
    private boolean statusAprov; //V
    private boolean statusConc; //M
    private double orca; //M
    private int idCliente; //V
    private int idPedido;
    private Pedido ListaPedidos;

    public Ordem(int indexPedido) {
        this.PedOrdem = ListaPedidos.getNomeProduto(indexPedido);
        this.idPedido = indexPedido;
        this.statusAprov = false;
        this.statusConc = false;
    }

    public String getPedOrdem() {
        return PedOrdem;
    }

    public void setPedOrdem(String PedOrdem) {
        this.PedOrdem = PedOrdem;
    }

    public String getRepMec() {
        return RepMec;
    }

    public void setRepMec(String RepMec) {
        this.RepMec = RepMec;
    }

    public boolean isStatusAprov() {
        return statusAprov;
    }

    public void setStatusAprov(boolean statusAprov) {
        this.statusAprov = statusAprov;
    }

    public boolean isStatusConc() {
        return statusConc;
    }

    public void setStatusConc(boolean statusConc) {
        this.statusConc = statusConc;
    }

    public double getOrca() {
        return orca;
    }

    public void setOrca(double orca) {
        this.orca = orca;
    }

    public int getIdCliente() {
        return idCliente;
    }

    public void setIdCliente(int idCliente) {
        this.idCliente = idCliente;
    }
    
    public int getIdPedido(){
        return this.idPedido;
    }
    
}
